Throughput Documentation Update (#6611)
Particular/docs.particular.net: nservicebus/usage/index.md

---
title: Measuring system usage
summary: Measure the usage of an NServiceBus system.
reviewed: 2024-05-08
related:
  - servicepulse/usage
  - nservicebus/throughput-tool
---

There are two methods provided for measuring usage of an NServiceBus system:

- [Using ServicePulse](#using-servicepulse)

> - [Standalone Endpoint throughput tool](#standalone-endpoint-throughput-tool)

Reviewer: ```suggestion
- [Standalone endpoint throughput tool](#standalone-endpoint-throughput-tool)
```

> [!NOTE]
> Requirements for both of the usage measuring methods:
>

> > - NServiceBus Transport: [Azure Service Bus](./../../transports/azure-service-bus), [RabbitMQ](./../../transports/rabbitmq), [Amazon SQS](./../../transports/sqs), [SqlServer](./../../transports/sql) OR

Reviewer: Probably safer to reference these with [/transports/rabbitmq] rather than relative references, no?

> - NServiceBus Transport: [MSMQ](./../../transports/msmq/) or [Azure Storage Queues](./../../transports/azure-storage-queues/) with [Auditing](./../operations/auditing.md) or [Monitoring](./../../monitoring/metrics) enabled on all NServiceBus endpoints

The output of the measuring methods is a usage report containing NServiceBus endpoint count and throughput summary. The report is generated as a JSON file which needs to be sent to Particular upon request, usually at license renewal time.

## Using ServicePulse

The recommended method of measuring the usage of an NServiceBus system is via [ServicePulse](/servicepulse/usage.md).

> [!NOTE]
> This method requires ServicePulse version 1.39 or later, and ServiceControl version 5.3 or later.

> This method provides the extra benefits of:

Reviewer: ```suggestion
Measuring throughput with ServicePulse offers the following advantages over the standalone throughput tool:
```

- ability to [view the system usage](/servicepulse/usage.md#viewing-usage-summary) at any time

> - ability to optionally [specify if a detected queue should not be included in license pricing](/servicepulse/usage.md#setting-an-endpoint-type)

Reviewer: ```suggestion
- ability to [exclude a detected queue from license pricing](/servicepulse/usage.md#setting-an-endpoint-type)
```

Author: Will leave this one as is as the queue is not excluded, but just marked with a reason as to why the customer would like to excluded it.

> - no requirement to run additional tools - [report is generated from a button click in ServicePulse](/servicepulse/usage.md#generating-a-usage-report)

Reviewer: ```suggestion
- no requirement to run additional tools - [the report can be generated directly from ServicePulse](/servicepulse/usage.md#generating-a-usage-report)
```

- improved endpoint identification

## Standalone Endpoint throughput tool

> Customers who are not able to use the latest ServiceControl version can use the [Endpoint throughput counter tool](./../throughput-tool) to measure their system usage.

Reviewer: ```suggestion
Customers who are not able to use a version of ServicePulse that supports throughput measurement can use the [endpoint throughput counter tool](./../throughput-tool) to measure their system usage.
```
